vulnfeeds/cmd/download-cves/main.go

```diff
@@ -74,26 +83,64 @@ func downloadCVE2(apiKey string, cvePath string) {
 	if err != nil {
 		log.Fatalf("Error reading response body: %+v", err)
 	}
-	var NVD2Data cves.NVDCVE2
+	var NVD2Data *cves.NVDCVE2
 	err = json.Unmarshal(body, &NVD2Data)
 	if err != nil {
 		log.Fatalf("Failed to decode NVD data: %+v", err)
 	}
-	Logger.Infof("%d results to download", NVD2Data.TotalResults)
-	// TODO: figure out how to paginate
-	file.Write(body)
+	Logger.Infof("At offset %d of total %d results", *NVD2Data.StartIndex, *NVD2Data.TotalResults)
+	return *NVD2Data
+}
+
+// Download all of the CVE data using the 2.0 API
+// See https://nvd.nist.gov/developers/vulnerabilities
+func downloadCVE2(APIKey string, CVEPath string) {
+	file, err := os.OpenFile(path.Join(CVEPath, "nvdcve-2.0.json.new"), os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0644)
+	defer file.Close()
+	if err != nil { // There's an existing file, check if it matches server file
+		Logger.Fatalf("Something went wrong when creating/opening file: %+v", err)
+	}
+	completeNVDdataset := cves.NVDCVE2{}
+	page := cves.NVDCVE2{}
+	offset := 0
+	for {
+		page = downloadCVE2WithOffset(APIKey, offset)
+		if len(page.Vulnerabilities) == 0 {
+			break
+		}
+		completeNVDdataset.Vulnerabilities = append(completeNVDdataset.Vulnerabilities, page.Vulnerabilities...)
+		offset += 2000
```

Comment:
Place page size in a const value

Reply:
Fixed in https://github.com/google/osv.dev/pull/773/commits/22653e9cf824f96eb7098b48e782f598bc6e8f01